app/src/main/java/nu/yona/app/ui/LaunchActivity.java

```diff
@@ -118,10 +122,71 @@ else if (!getSharedUserPreferences().getBoolean(PreferenceConstant.STEP_PASSCODE
 		}
 		else if (!TextUtils.isEmpty(getSharedUserPreferences().getString(PreferenceConstant.YONA_PASSCODE, "")))
 		{
-			startNewActivity(bundle, YonaActivity.class);
+			checkForUserEntityVersion();// validate any updates to user entity before moving to next screen.
+		}
+	}
+
+	private void checkForUserEntityVersion()
+	{
+		try
+		{
+			JSONObject userJsonObj = APIManager.getInstance().getAuthenticateManager().getStoredUserObjectFromDB();
+			if (userJsonObj.isNull("version") || userJsonObj.getInt("version") != AppConstant.USER_ENTITY_VERSION)
+			{
+				if (!AppUtils.isNetworkAvailable(YonaApplication.getAppContext()))
+				{
+					showMandatoryUserFetchError();
+				}
+				else
+				{
+					JSONObject userLinks = userJsonObj.getJSONObject("links");
+					JSONObject userSelfHrefObj = userLinks.getJSONObject("self");
+					String userSelfLink = userSelfHrefObj.getString("href");
+					getUserFromServer(userSelfLink);
+				}
+			}
+			else
+			{
+				moveToYonaActivity();
+			}
+		}
+		catch (JSONException e)
+		{
+			showMandatoryUserFetchError();
 		}
 	}
 
+	private void moveToYonaActivity()
+	{
+		startNewActivity(bundle, YonaActivity.class);
+		YonaAnalytics.trackCategoryScreen(AnalyticsConstant.LAUNCH_ACTIVITY, AnalyticsConstant.LAUNCH_ACTIVITY);
+	}
+
+	private void getUserFromServer(String userSelfLink)
+	{
+		showLoadingView(true);
+		DataLoadListenerImpl dataLoadListenerImpl = new DataLoadListenerImpl((result) -> handleUserFetchSuccess(result), (result) -> handleUserFetchFailure(result), null);
+		APIManager.getInstance().getAuthenticateManager().getUserFromServer(userSelfLink, dataLoadListenerImpl);
+	}
+
+	private Object handleUserFetchSuccess(Object result)
+	{
+		moveToYonaActivity();
+		showLoadingView(false);
+		return null;
```

Comment:
Add the explanatory comment we have as convention.

Reply:
Done.